src/hotspot/os/linux/systemMemoryBarrier_linux.cpp

```diff
@@ -56,12 +56,12 @@ enum membarrier_cmd {
   MEMBARRIER_CMD_REGISTER_PRIVATE_EXPEDITED = (1 << 4),
 };
 
-static int membarrier(int cmd, unsigned int flags, int cpu_id) {
+static int64_t membarrier(int cmd, unsigned int flags, int cpu_id) {
```

Comment:
Seems OK, but if syscall returns `long` then intrptr_t seems a better fit for 32-bit platforms.  Or we could narrow to int here, as long as membarrier (and futex) aren't returning pointers or memory addresses.

Reply:
I think intptr_t is a better choice.